torchci/lib/bot/retryBot.ts

```diff
@@ -58,54 +145,71 @@ function retryBot(app: Probot): void {
       return;
     }
 
-    // @ts-expect-error - we don't have types for these
-    let doesLookLikeUserFailure = (job, isCodeValiationStep: (step: any) => boolean) => {
+    const doesLookLikeUserFailure = (
```

Comment:
Refactor the main function to make it easier to follow

Reply:
There are now only 2 functions `await retryPreviousWorkflow` and `await retryCurrentWorkflow` in the main body.